include/wx/srchctrl.h

```diff
@@ -25,8 +25,14 @@
 
     class WXDLLIMPEXP_CORE wxSearchCtrlBaseBaseClass
         : public wxCompositeWindow< wxNavigationEnabled<wxControl> >,
-          public wxTextEntryBase
+          public wxTextEntry
     {
+#if defined(__WXMSW__)
+    private:
+        // implement this to return the HWND of the EDIT control
+        // can return nullptr here as this TextEntry is just a proxy
+        virtual WXHWND GetEditHWND() const override  { return nullptr; }
```

Comment:
I do not think that is necessary (I try to avoid this because it would mean adding WXMSW specific code to a generic source file) because this is a private method **exclusively** used in the wxTextEntry implementation itself. The generic wxSearchCtrl implementation however overrides and routes **all** public wxTextEntry methods to the m_text member which has it's own GetHwnd() implementation.
BTW I have tested the wxTextEntry interface after this change on wxRuby3 on windows and it worked (I'll look at adding wxTextEntry tests to the wxSearchCtrl regression test though).

Reply:
Hmm, yes, sorry, you're right. It's still somewhat uncomfortable to have this "clearly wrong" (even if it isn't, on closer examination) implementation. I wonder if we could put `wxFAIL_MSG("unreachable")` in it?

Reply:
That should be fine too. If all checks out (and it should) this method  should never be called. I'll make the addition.

Reply:
Done.